internal/grpc/grpc.go

```diff
@@ -34,11 +60,75 @@ var (
 	}`
 
 	defaultCredentials = insecure.NewCredentials()
+
+	_ GrpcConnectionInterface = (*GrpcConnection)(nil)
 )
 
-type wrappedStream struct {
-	grpc.ClientStream
-	*protovalidate.Validator
+func NewGrpcConnection(ctx context.Context, agentConfig *config.Config) (*GrpcConnection, error) {
+	if agentConfig == nil || agentConfig.Command.Server.Type != "grpc" {
```

Comment:
We should probably have an enum for `agentConfig.Command.Server.Type` at some point.

Reply:
good point let me see what I can do